e2e-tests/tfm/cypress/e2e/journeys/deal-cancellation/check-details.spec.js

```diff
@@ -99,60 +107,136 @@ context('Deal cancellation - check details', () => {
+    describe('effective from date change link', () => {
+      it('takes you to the effective from page and correctly updates any changes', () => {
+        const testReason = 'test reason';
+        checkDetailsPage.effectiveFromLink().click();
 
-      cy.url().should('eq', relative(`/case/${dealId}/cancellation/effective-from-date`));
+        cy.url().should('eq', relative(`/case/${dealId}/cancellation/effective-from-date?status=change`));
 
-      cy.keyboardInput(effectiveFromDatePage.effectiveFromDateDay(), dateConstants.threeMonthsOneDayDay);
-      cy.keyboardInput(effectiveFromDatePage.effectiveFromDateMonth(), dateConstants.threeMonthsOneDayMonth);
-      cy.keyboardInput(effectiveFromDatePage.effectiveFromDateYear(), dateConstants.threeMonthsOneDayYear);
+        cy.keyboardInput(effectiveFromDatePage.effectiveFromDateDay(), dateConstants.threeMonthsOneDayDay);
+        cy.keyboardInput(effectiveFromDatePage.effectiveFromDateMonth(), dateConstants.threeMonthsOneDayMonth);
+        cy.keyboardInput(effectiveFromDatePage.effectiveFromDateYear(), dateConstants.threeMonthsOneDayYear);
 
-      cy.clickContinueButton();
+        cy.clickContinueButton();
 
-      cy.url().should('eq', relative(`/case/${dealId}/cancellation/check-details`));
-      cy.assertText(checkDetailsPage.reasonResponse(), testReason);
-      cy.assertText(checkDetailsPage.bankRequestDateResponse(), format(dateConstants.today, 'd MMMM yyyy'));
-      cy.assertText(checkDetailsPage.effectiveFromResponse(), format(dateConstants.threeMonthsOneDay, 'd MMMM yyyy'));
+        cy.url().should('eq', relative(`/case/${dealId}/cancellation/check-details`));
+        cy.assertText(checkDetailsPage.reasonResponse(), testReason);
+        cy.assertText(checkDetailsPage.bankRequestDateResponse(), format(dateConstants.today, 'd MMMM yyyy'));
+        cy.assertText(checkDetailsPage.effectiveFromResponse(), format(dateConstants.threeMonthsOneDay, 'd MMMM yyyy'));
+      });
+
+      it('takes you to the effective from page and pressing "Back" returns you to the summary page', () => {
+        checkDetailsPage.effectiveFromLink().click();
+
+        cy.url().should('eq', relative(`/case/${dealId}/cancellation/effective-from-date?status=change`));
+
+        cy.clickBackLink();
+
+        cy.url().should('eq', relative(`/case/${dealId}/cancellation/check-details`));
+      });
+
+      it('takes you to the effective from page and pressing "Back" returns you to the summary page after validation errors', () => {
+        checkDetailsPage.effectiveFromLink().click();
+        cy.url().should('eq', relative(`/case/${dealId}/cancellation/effective-from-date?status=change`));
+
+        cy.keyboardInput(effectiveFromDatePage.effectiveFromDateYear(), '1');
+        cy.clickContinueButton();
+        errorSummary();
+
+        cy.clickBackLink();
+
+        cy.url().should('eq', relative(`/case/${dealId}/cancellation/check-details`));
+      });
```

Comment:
I'm finding these tests a bit hard to read, especially as there are multiple pages and scenarios involved and I don't have the context of the full journey; I think we should split these up. Something like this, what do you think?

```js
describe("'x row' - change link", () => {
  beforeEach(() => {
    checkDetailsPage.bankRequestDateLink().click();
  });

  it('should redirect to the `bank request date` page', () => {
    cy.url().should('eq', relative(`/case/${dealId}/cancellation/bank-request-date?status=change`));
  });

  describe('when submitting X form', () => {
    it('should render updates', () => {
      // cy.keyboardInput(...)
      // cy.assertText(...)
    });
  });

  describe('when clicking the back link', ( ) => {
    it('should redirect to X page', () => {
    });
  });

  describe('when clicking the back link with validation errors', () => {
    it('should redirect to X page', () => {
    });

    it('should NOT render updates', () => {
    });
  });
});
```

Reply:
Good idea, have now split them out as suggested - have a look and see what you think!